legend-depot-store-metrics/src/main/java/org/finos/legend/depot/store/metrics/services/QueryMetricsHandler.java

```diff
@@ -58,6 +59,15 @@ public List<VersionQueryMetric> getSummaryByProjectVersion()
         return all.collect(Collectors.toList());
     }
 
+    public List<VersionQueryMetric> findLeastRecentlyUsedMetrics(boolean isSnapshot, Date date)
+    {
+        if (isSnapshot)
+        {
+            return metricsStore.findLeastRecentlyUsed("-SNAPSHOT$", date);
+        }
+        return metricsStore.findLeastRecentlyUsed(".*\\..*\\..*", date);
```

Comment:
this implementation is not going to scale very well plus the intention of the method is not what the name does.. as per my comment above.